src/SimpleThings/EntityAudit/AuditConfiguration.php

```diff
@@ -27,15 +27,49 @@
 
 class AuditConfiguration
 {
-    private $prefix = '';
-    private $suffix = '_audit';
-    private $revisionFieldName = 'rev';
-    private $revisionTypeFieldName = 'revtype';
-    private $revisionTableName = 'revisions';
-    private $auditedEntityClasses = array();
-    private $globalIgnoreColumns = array();
-    private $currentUsername = '';
-    private $revisionIdFieldType = 'integer';
+    private $auditedEntityClasses;
+    private $globalIgnoreColumns;
+    private $tablePrefix;
+    private $tableSuffix;
+    private $revisionTableName;
+    private $revisionFieldName;
+    private $revisionTypeFieldName;
+    private $revisionIdFieldType;
+    private $usernameCallable;
+
+    /**
+     * @param array $auditedEntityClasses
+     * @param array $globalIgnoreColumns
+     * @param $tablePrefix
+     * @param $tableSuffix
+     * @param $revisionTableName
+     * @param $revisionFieldName
+     * @param $revisionTypeFieldName
+     * @param $revisionIdFieldType
+     * @param callable $usernameCallable
+     */
+    public function __construct(
```

Comment:
Instead of using `__construct` here, I suggest moving the defaults back into the properties, and then defining a named constructor. The reason for doing so is that additional fields will always come up, and sometimes you may want to make them required. In order to do so, you do not want to change `__construct` (which would result in a BC break), but deprecate the old named constructor and add a new one.